qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -0,0 +1,313 @@
+
+        self.add_parameter(
+            "open_state",
+            get_cmd=":CORRection:OPEN:STATe?",
+            set_cmd=":CORRection:OPEN:STATe {}",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0"),
+            docstring="Enables or disable OPEN correction"
+        )
+
+        self.add_parameter(
+            "short",
+            set_cmd=":CORRection:SHORt",
+            docstring="Executes SHORT correction based on all frequency points."
+        )
+
+        self.add_parameter(
+            "short_state",
+            get_cmd=":CORRection:SHORt:STATe?",
+            set_cmd=":CORRection:SHORt:STATe {}",
+            val_mapping=create_on_off_val_mapping(on_val="1", off_val="0"),
+            docstring="Enables or disable SHORT correction."
+        )
+
+
+class KeysightE4980A(VisaInstrument):
+    """
+    QCodes driver for E4980A Precision LCR Meter
+    """
+    def __init__(self, name, address, terminator='\n', **kwargs):
+        """
+        Create an instance of the instrument.
+
+        Args:
+            name: Name of the instrument instance
+            address: Visa-resolvable instrument address.
+        """
+        super().__init__(name, address, terminator=terminator, **kwargs)
+
+        self._measurement_pair = MeasurementPair(
+            "CPD",
+            ("capacitance", "dissipation_factor"),
+            ("F", "")
+        )
+
+        self.add_parameter(
+            "frequency",
+            get_cmd=":FREQuency?",
+            set_cmd=":FREQuency {}",
+            get_parser=float,
+            unit="Hz",
+            vals=Numbers(20, 2E6),
+            docstring="Gets and sets the frequency for normal measurement."
+        )
+
+        self.add_parameter(
+            "current_level",
+            get_cmd=":CURRent:LEVel?",
+            set_cmd=":CURRent:LEVel {}",
+            get_parser=float,
+            unit="A",
+            vals=Numbers(0, 0.1),
+            docstring="Gets and sets the current level for measurement signal."
+        )
+
+        self.add_parameter(
+            "voltage_level",
+            get_cmd=":VOLTage:LEVel?",
+            set_cmd=":VOLTage:LEVel {}",
+            get_parser=float,
+            unit="V",
+            vals=Numbers(0, 20),
+            docstring="Gets and sets the voltage level for measurement signal."
+        )
+
+        self.add_parameter(
+            "measurement_function",
+            get_cmd=":FUNCtion:IMPedance?",
+            set_cmd=self._set_measurement
+        )
+
+        self.add_parameter(
+            "range",
+            get_cmd=":FUNCtion:IMPedance:RANGe?",
+            set_cmd=":FUNCtion:IMPedance:RANGe {}",
+            unit='Ohm',
+            vals=Enum(0.1, 1, 10, 100, 300, 1000, 3000, 10000, 30000, 100000),
+            docstring="Selects the impedance measurement range, also turns "
+                      "the auto range function OFF."
+        )
+
+        self.add_parameter(
+            "system_errors",
```

Comment:
please make this a bound method as opposed to a parameter - "getting errors" is not a feature that is meant to be wrapped into a parameter

Reply:
Fixed. now it's a method instead of a parameter.